### PR TITLE
ENH regularize T for shape

### PR DESCRIPTION
**Pull request by beckermr**
Trying a different way to smooth the shapes.

### CODE REVIEW

File: metadetect/lsst/measure.py [hunk 1]
```diff
@@ -179,6 +180,9 @@ def measure(
         For calculating moments
     stamp_size: int
         Size for postage stamps
+    fwhm_reg: int, optional
```
Comment (beckermr) [inline metadetect/lsst/measure.py:183]: should be `float` here.

**Comment by esheldon** [comment]
Another option would be to have the fitters put this into their sums as a `Tsums_reg` so we don't have to pass around this extra parameter to functions

**Comment by beckermr** [comment]
I can't stomach more global changes to ngmix.

File: metadetect/fitting.py [hunk 2]
```diff
@@ -438,8 +457,28 @@ def _sum_bands_wavg(
...
+
+        reg_mom = raw_mom.copy()
+        reg_mom_cov = raw_mom_cov.copy()
+        # the moments are not normalized and are sums, so convert T_reg to a sum using
+        # the flux sum first via T_reg -> T_reg * raw_mom[5]
+        reg_mom[4] = raw_mom[4] + T_reg * raw_mom[5]
+        # TODO double check this math
+        reg_mom_cov[4, 4] = raw_mom_cov[4, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[4, 5] = reg_mom_cov[4, 5] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[5, 4] = reg_mom_cov[5, 4] + T_reg**2 * raw_mom_cov[5, 5]
```
Comment (beckermr) [inline metadetect/fitting.py:474]:
```suggestion
        reg_mom_cov[4, 5] = raw_mom_cov[4, 5] + T_reg**2 * raw_mom_cov[5, 5]
        reg_mom_cov[5, 4] = raw_mom_cov[5, 4] + T_reg**2 * raw_mom_cov[5, 5]
```

File: metadetect/fitting.py [hunk 3]
```diff
@@ -438,8 +457,28 @@ def _sum_bands_wavg(
...
+        reg_mom[4] = raw_mom[4] + T_reg * raw_mom[5]
+        # TODO double check this math
+        reg_mom_cov[4, 4] = raw_mom_cov[4, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[4, 5] = raw_mom_cov[4, 5] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[5, 4] = raw_mom_cov[5, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        momres = make_mom_result(reg_mom, raw_mom_cov)
+
+        # use old T
+        for col in ["T", "T_err", "T_flags", "T_flagstr"]:
+            momres[col] = momres_t[col]
```
Comment (beckermr) [inline metadetect/fitting.py:479]:
```suggestion
            momres[col] = momres_t[col]

        momres["flags"] |= momres_t["flags"]
        momres["flagstr"] = ngmix.flags.get_flags_str(momres["flags"])

```

File: metadetect/fitting.py [hunk 4]
```diff
@@ -438,8 +457,32 @@ def _sum_bands_wavg(
...
+        reg_mom = raw_mom.copy()
+        reg_mom_cov = raw_mom_cov.copy()
+        # the moments are not normalized and are sums, so convert T_reg to a sum using
+        # the flux sum first via T_reg -> T_reg * raw_mom[5]
+        reg_mom[4] = raw_mom[4] + T_reg * raw_mom[5]
+        # TODO double check this math
+        reg_mom_cov[4, 4] = raw_mom_cov[4, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[4, 5] = raw_mom_cov[4, 5] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[5, 4] = raw_mom_cov[5, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        momres = make_mom_result(reg_mom, raw_mom_cov)
```
Comment (beckermr) [inline metadetect/fitting.py:475]:
```suggestion
        momres = make_mom_result(reg_mom, reg_mom_cov)
```

File: metadetect/fitting.py [hunk 5]
```diff
@@ -438,8 +457,32 @@ def _sum_bands_wavg(
...
+        # TODO double check this math
+        reg_mom_cov[4, 4] = raw_mom_cov[4, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[4, 5] = raw_mom_cov[4, 5] + T_reg**2 * raw_mom_cov[5, 5]
+        reg_mom_cov[5, 4] = raw_mom_cov[5, 4] + T_reg**2 * raw_mom_cov[5, 5]
+        momres = make_mom_result(reg_mom, reg_mom_cov)
+
+        # use old T
+        for col in ["T", "T_err", "T_flags", "T_flagstr"]:
+            momres[col] = momres_t[col]
+
```
Comment (beckermr) [inline metadetect/fitting.py:480]:
```suggestion
        # the moments are not normalized and are sums, so convert T_reg to a sum using
        # the flux sum first via T_reg -> T_reg * raw_mom[5]
        amat = np.eye(6)
        amat[4, 5] = T_reg
        reg_mom = np.dot(amat, raw_mom)
        reg_mom_cov = np.dot(amat, np.dot(raw_mom_cov, amat.T))
        momres = make_mom_result(reg_mom, reg_mom_cov)

        # use old T
        for col in ["T", "T_err", "T_flags", "T_flagstr"]:
            momres[col] = res[col]

        momres["flags"] |= res["flags"]
        momres["flagstr"] = ngmix.flags.get_flags_str(momres["flags"])

```

**Comment by beckermr** [comment]
@esheldon This one is ready for review!

**Pull request merged**